File: src/lhs/C/lhss.c
/* lhss.c */
/* Spec test for the LHS C interface */

#include <stdio.h>
#include <math.h>
#include "lhs.h"

int main(void) {

    // Derived types
    void *data;
    struct lhs_control_type control;
    struct lhs_inform_type inform;

    // Initialize LHS
    lhs_initialize(&data, &control, &inform);

    // Parameters
    int n_dimen = 7; // dimension
    int n_points = 2; // points required
    int X[n_dimen][n_points]; // points
    int seed;
    
    // Set a random seed
    lhs_get_seed(&seed);                             

    // Generate points
    lhs_ihs(n_dimen, n_points, &seed, X, &control, &inform, &data);
    if(inform.status == 0){ // successful return
        printf("LHS successful\n");
        for(int j = 0; j < n_points; j++){
            printf("Point %d = ", j);
            for(int i = 0; i < n_dimen; i++){
                printf("%d ", X[i][j]);
            }
            printf("\n");
        }
    }else{ // error returns
        printf("LHS exit status = %d \n", inform.status);
    }

    // Delete internal workspace
    lhs_terminate(&data, &control, &inform);

    return 0;
}

File: src/lhs/lhss.f90
   PROGRAM GALAHAD_LHS_EXAMPLE  !  GALAHAD 2.8 - 29/06/2016 AT 15:40 GMT
   USE GALAHAD_LHS_double       ! double precision version
   IMPLICIT NONE
   INTEGER, PARAMETER :: wp = KIND( 1.0D+0 )                ! set precision
   TYPE ( LHS_control_type ) :: control
   TYPE ( LHS_inform_type ) :: inform
   TYPE ( LHS_data_type ) :: data
   INTEGER, PARAMETER :: n_dimen = 7                        ! dimension
   INTEGER, PARAMETER :: n_points = 2                       ! # points required
   INTEGER :: X( n_dimen, n_points )                        ! points
   INTEGER :: j, seed
   CALL LHS_initialize( data, control, inform )             ! set controls
   CALL LHS_get_seed( seed )                                ! set a random seed
   CALL LHS_ihs( n_dimen, n_points, seed, X,                                   &
                 control, inform, data )                    ! generate points
   IF ( inform%status == 0 ) THEN                           ! Successful return
    DO j = 1, n_points
      WRITE( 6, "( ' point ', I0, ' =', 7I3 )" ) j, X( : , j )
    END DO
   ELSE                                                      ! Error returns
     WRITE( 6, "( ' LHS_ihs exit status = ', I6 ) " ) inform%status
   END IF
   CALL LHS_terminate( data, control, inform )    ! deallocate workspace arrays
   END PROGRAM GALAHAD_LHS_EXAMPLE
